### PR TITLE
Clippy Book Chapter Updates Reborn: Lint Passes

### PR DESCRIPTION
**Pull request by blyxyas**
This PR adds a new chapter to the book: "Lint passes". No major changes apart from some re-phrasing, fixing typos... etc.

## Notes

- Requires #10595 to be merged before this one (Or else, a link will be broken).
- To talk about the whole project, please use the tracking issue for the project #10597 (It also contains a timeline, discussions and more information)

changelog: Add a new "Lint passes" chapter to the book
r? @flip1995

### CODE REVIEW

File: book/src/development/lint_passes.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
+# Lint passes
+
+Before working on the logic of a new lint, there is an important decision
+that every Clippy developers must make: to use
+[`EarlyLintPass`][early_lint_pass] or [`LateLintPass`][late_lint_pass].
+
+In short, the `LateLintPass` has access to type information while the
```
Comment (llogiq) [inline book/src/development/lint_passes.md:7]: Not only type information but also `resolve`: In an early pass, you usually have no information what a symbol means, so most early lints are plainly syntactic.

Comment (blyxyas) [reply book/src/development/lint_passes.md]: Fixed. (Talks about that in the introduction and in the "Early Lint Pass" section)

**Comment by llogiq** [comment]
Given that @flip1995 still seems quite busy, I think he'll be ok if I take over the review for this chapter, too, so...thank you!

@bors r+

**Comment by flip1995** [comment]
Thanks, I have a 12h flight ahead of me on Sunday, where I'll look at these PRs.

**Pull request merged**